include/swift/AST/DiagnosticsParse.def

```diff
@@ -1500,6 +1500,8 @@ ERROR(attr_differentiable_missing_label,PointsToFirstBadToken,
 ERROR(attr_differentiable_expected_label,none,
       "expected either 'wrt:' or a function specifier label, e.g. 'jvp:', "
       "or 'vjp:'", ())
+ERROR(attr_differentiable_no_vjp_or_jvp_when_linear,none,
+      "can't define 'vjp:' and/or 'jvp:' on functions marked as 'linear'", ())
```

Comment:
I might recommend preemptively mentioning `'transpose:'` (even though it isn't implemented yet).

Something along the lines of (not perfect):
```suggestion
      "cannot specify 'vjp:' or 'jvp:' for linear functions; use 'transpose:' instead", ())
```
cc @rxwei

Reply:
Ya that's a good idea we can just add it in right away. 😄